File: Week 07/id_662/231.power-of-two.java
/**
 * 给定一个整数，编写一个函数来判断它是否是 2 的幂次方。
 * <p>
 * 示例 1:
 * <p>
 * 输入: 1
 * 输出: true
 * 解释: 20 = 1
 * 示例 2:
 * <p>
 * 输入: 16
 * 输出: true
 * 解释: 24 = 16
 * 示例 3:
 * <p>
 * 输入: 218
 * 输出: false
 * <p>
 */

class Solution {
    public boolean isPowerOfTwo(int n) {
        return n > 0 && (n & (n - 1)) == 0;
    }
}

File: Week 07/id_377/231.power-of-two.go
/*
 * @lc app=leetcode id=231 lang=golang
 *
 * [231] Power of Two
 * 
 */

// @lc code=start


func isPowerOfTwo(n int) bool {
	if n < 1 {
		return false
	}

	for n > 1 {
		if n%2 == 1 {
			return false
		}
		n /= 2
	}

	return true
}

 
//  var isPowerOfTwo = function(n) {
// 	var tmp = ~~(Math.log(n) / Math.log(2));
// 	return n === (1 << tmp);
//   };

/* class Solution(object):
    def isPowerOfTwo(self, n):
        """
        :type n: int
        :rtype: bool
        """
        return n != 0 and (n & -n) == n */

// @lc code=end

